<launch>
  <rosparam param="/use_sim_time">true</rosparam>
  <!-- Time Surface (TS) generation -->
  <node name="TimeSurface_left" pkg="ts_ros" type="time_surface" output="screen">
    <!-- <remap from="events" to="/davis_left/events" /> -->
    <!-- <remap from="events" to="/davis/left/events" /> -->
    <remap from="events" to="/davis346/events" />
    <remap from="image" to="/davis_left/image_raw" />
    <!-- <remap from="image" to="/davis/left/image_mono" /> -->
    <remap from="camera_info" to="/davis_left/camera_info" />
    <remap from="time_surface" to="TS_left" />

    <rosparam command="load" file="$(find ts_ros)/cfg/parameters.yaml" />
  </node>

  <node name="TimeSurface_right" pkg="ts_ros" type="time_surface" output="screen">
    <remap from="events" to="/davis_right/events" />
    <remap from="image" to="/davis_right/image_raw" />
    <remap from="camera_info" to="/davis_right/camera_info" />
    <remap from="adaptive_time_surface" to="TS_right" />

    <rosparam command="load" file="$(find ts_ros)/cfg/parameters.yaml" />
  </node>

  <!-- Synchronize both TS processing using an external timer -->
  <node name="global_timer" pkg="rostopic" type="rostopic" args="pub -s -r 100 /sync std_msgs/Time 'now' " />

  <!-- Visualization -->
  

</launch>
